### PR TITLE
Move to RSpec tests

### PR DESCRIPTION
**Pull request by zubin**
### Context
The tests are currently written in Minitest. We use RSpec everywhere else. 

### Change
- Port the tests over to RSpec. 
- Remove Minitest.

### CODE REVIEW

File: spec/fixtures/daily.bai2 [hunk 1]
```diff
@@ -1,8 +1,8 @@
...
-49,25001,4/
-98,25001,1,6/
+01,121140399,9999999999,050608,1015,1,80,1,2/
+02,9999999999,121140399,1,050607,1015,,2/
+03,1234567890,,,,,/
+16,174,25001,Z,,50848,/
+88,SAMPLE CONTINUATION TEXT,/
+49,25001,4/
+98,25001,1,6/
 99,25001,1,8/
```
Comment (kieranjogrady) [inline spec/fixtures/daily.bai2]: Should we add newline at end of file?
```suggestion
99,25001,1,8/

```

File: spec/support/fixtures.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+module Fixtures
+  FIXTURE_PATH = Pathname.new(File.expand_path("../fixtures", __dir__))
+  private_constant :FIXTURE_PATH
+
+  def self.path(path)
+    FIXTURE_PATH / path
```
Comment (kieranjogrady) [inline spec/support/fixtures.rb:8]: Ooh, nice one 😎. I haven't seen `#/` used like this before. For anyone else that is curious, [Pathname#/](https://ruby-doc.org/stdlib-3.0.2/libdoc/pathname/rdoc/Pathname.html#method-i-2B) is an alias for `Pathname#+`.

**Review by kieranjogrady — APPROVED** [review]
Nice work. If we're aligning the gem more with our own standards and conventions compared to the upstream repo from where it was forked should we consider using proper uppercase acronyms for the `BAI2` and `BAI` modules/classnames? We could probably even do it in a way that maintains backwards compatibility if there is an interest in submitting a PR to the upstream project. WDYT?

**Comment by zubin** [comment]
> Nice work. If we're aligning the gem more with our own standards and conventions compared to the upstream repo from where it was forked should we consider using proper uppercase acronyms for the `BAI2` and `BAI` modules/classnames? We could probably even do it in a way that maintains backwards compatibility if there is an interest in submitting a PR to the upstream project. WDYT?

I don't have strong opinions on inflections. I find them useful for things which end up being rendered in rails views (eg a form builder might apply `bai_file.humanize` for an input label; configuring an inflection negates the need to set a `label` kwarg). 

Wei seemed to think the upstream project was abandoned (no updates for 2 years, no response to his issue).

File: spec/fixtures/daily.bai2 (hunk 1 above)
Comment (zubin) [reply spec/fixtures/daily.bai2]: Not sure, I didn't intentionally change anything. Is there meant to be one there?

File: spec/support/fixtures.rb (hunk 2 above)
Comment (zubin) [reply spec/support/fixtures.rb:8]: I didn't know about `+`, I thought it was an alias for `#join`. Python uses `/`.

**Pull request merged**